fgpyo/sam/__init__.py

```diff
@@ -531,6 +531,15 @@ def __str__(self) -> str:
         else:
             return "*"
 
+    def __len__(self) -> int:
+        return len(self.elements)
+
+    def __iter__(self) -> Iterator[CigarElement]:
+        return iter(self.elements)
+
+    def __getitem__(self, index: int) -> CigarElement:
+        return self.elements[index]
+
```

Comment:
I love this.  Can we add to the module-level examples to demonstrate this new functionality?